locations/descriptor_locations.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package locations
+
+import (
+	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// DescriptorName returns the precise location for a descriptor's name.
+//
+// This works for any descriptor, regardless of type (message, field, etc.).
```

Comment:
Does it work for `FileDescriptor`? I guess not. But I think `FileDescriptor` will return `nil` for `SourceInfo`, so the code should work as normal. Can you add a test for `FileDescriptor` as well? Thanks!

Reply:
Yes, `FileDescriptor` [has a name, and it is number 1](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto#L63).
I have no clue what the corresponding `Span` is, though.

Reply:
So the path for the name of the `FileDescriptor` will become `[1]`. And I think there is no `SourceInfo` for the path `[1]`, so it should return `nil`. Can you double check it? Thanks!

Reply:
Correct, it returned nil.